mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala

```diff
@@ -11,12 +13,20 @@ import org.backuity.clist.{Command, _}
  */
 class CheckCmd extends Command(name = "check", description = "Check a TLA+ specification") with General {
 
+  // Parses the smtEncoding option
+  implicit val smtEncodingRead: Read[SMTEncoding] =
+    Read.reads[SMTEncoding]("a SMT encoding, either oopsla19 or arrays") {
+      case "arrays"        => arraysEncoding
+      case "oopsla19"      => oopsla19Encoding
+      case oddEncodingType => throw new IllegalArgumentException(s"Unexpected SMT encoding type $oddEncodingType")
+    }
+
```

Comment:
This is actually really lovely. It introduces a clear example of the right way to do all such flags I think, and sets a great precedent. Thanks for doing the work to get this in place!